flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1325,26 +1497,38 @@ def run_to_convergence(self, desired_datasets, max_iterations=20):
             one desired state to another if there are no bugs.
         """
         for _ in range(max_iterations):
-            self.run_convergence_step(
-                dataset_map_from_iterable(desired_datasets))
-            note(self.current_datasets())
-            if compare_dataset_states(
-                self.current_datasets(),
-                dataset_map_from_iterable(desired_datasets),
-            ):
-                break
+            try:
+                self.run_convergence_step(
+                    dataset_map_from_iterable(desired_datasets))
+                if compare_dataset_states(
+                    self.current_datasets(),
+                    dataset_map_from_iterable(desired_datasets),
+                ):
+                    break
+            except Exception as e:
```

Comment:
Yeah, I'd be happier without this. But due [this](https://github.com/ClusterHQ/flocker/pull/2571#discussion_r53838436) we actually trigger this, trying to attach or detach a volume that isn't in an appropriate state.